web/src/hooks/useLikePost.ts

```diff
@@ -19,7 +19,7 @@ export default function useLikePost({
 
   const { mutate: mutateLike, isLoading: isLiking } = useMutation({
     mutationKey: ["like"],
-    mutationFn: () => likePost(postId),
+    mutationFn: () => likePost(postId ?? ""),
```

Comment:
No, so this just moves the same issue down a level. These functions shouldn't be interpreting `""` as a no-op imo. I would favor
```ts
mutationFn: () => {
  if (postId) likePost(postId)
}
```

Reply:
I can move it down another level

Reply:
My point is just that at the first level where we know there is no `postId` and it's convenient, we should stop calling things.

Reply:
Tbh, I think the `if` should go in `mutationFn`